src/Microsoft.Azure.WebJobs/FunctionNameAttribute.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Azure.WebJobs
+{
+    /// <summary>
+    /// This is for providing a custom name to for an SDK function.
```

Comment:
This was meant to completely replace the name the host is using when referring to this function. As I said I only added that because it looked simple enough, but if you think it's more work to get it working end to end, then I will remove it and leave just the Attribute for the tooling to consume.